### PR TITLE
refactor: socket logic from Chat.jsx

### LINKED ISSUE #664: [OTHER] move all socket function in chat.jsx into chatSocket.js
### What would you like to share?

The chat component
https://github.com/Dun-sin/Whisper/blob/dbec99c68f63c606647ea2cef1d458a72ae83b1b/client/src/components/Chat.jsx

The chatSocket file:
https://github.com/Dun-sin/Whisper/blob/dbec99c68f63c606647ea2cef1d458a72ae83b1b/client/src/lib/chatSocket.js

### PR DESCRIPTION
**Pull request by damirgros**
# Fixes Issue

**My PR closes #664**

# 👨‍💻 Changes proposed(What did you do ?)

Moved "emitTyping" function,  socket logic and handlers from "Chat.jsx" to "chatSocket.js".

Adjusted all imports.

# ✔️ Check List (Check all the applicable boxes)

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

# 📷 Screenshots

### CODE REVIEW

**Comment by damirgros** [comment]
My bad. I pushed the changes.

File: client/src/lib/chatSocket.js [hunk 1]
```diff
@@ -94,10 +124,73 @@ export default function useChatUtils(socket) {
...
+		socket.on(NEW_EVENT_SEND_FAILED, onLimitMessageHandler);
+
+		return () => {
+			socket.off(NEW_EVENT_RECEIVE_MESSAGE, onNewMessageHandler);
+			socket.off(NEW_EVENT_DELETE_MESSAGE, onDeleteMessageHandler);
+			socket.off(NEW_EVENT_EDIT_MESSAGE, onEditMessageHandler);
+			socket.off(NEW_EVENT_READ_MESSAGE, onReadMessageHandler);
+			socket.off(NEW_EVENT_SEND_FAILED, onLimitMessageHandler);
+			socket.off('publicKey', onPublicStringHandler);
+		};
```
[the remark this answers is not in the document]
Comment (damirgros) [reply client/src/lib/chatSocket.js]: Pushed the changes.
I'm getting no errors in dev.

**Comment by Dun-sin** [comment]
Please provide a screen recording to show everything is working fine

**Comment by Dun-sin** [comment]
hi @damirgros you have 2 more days to perform the requested change(s) after that your PR will be closed

**Pull request closed without merging**